h5p-bildetema/src/components/SearchPage/ChooseCollectionDialog/ChooseCollectionDialog.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+import { useMyCollections } from "common/hooks/useMyCollections";
+import Dialog from "../../Dialog/Dialog";
+import Button from "../../Button/Button";
+import Select from "../../Select/Select";
+import TextInput from "../../TextInput/TextInput";
+import { AddIcon } from "../../Icons/Icons";
+import styles from "./ChooseCollectionDialog.module.scss";
+
+type CollectionOption = {
+  label: string;
+};
+
+export type ChooseCollectionDialogProps = {
+  open: boolean;
+  options: CollectionOption[];
+  selected: CollectionOption | null;
+  selectedWordId: string | null;
+  setSelected: (option: CollectionOption) => void;
+  onClose: () => void;
+  onChange: (option: CollectionOption) => void;
+  onAddBookmark: () => void;
+};
+
+const ChooseCollectionDialog = ({
+  open,
+  options,
+  selected,
+  selectedWordId,
+  setSelected,
+  onClose,
+  onChange,
+  onAddBookmark,
+}: ChooseCollectionDialogProps): JSX.Element => {
+  const { addItemToCollection, addCollection } = useMyCollections();
+
+  const [createNewCollection, setCreateNewCollection] = useState(false);
+  const [textInput, setTextInput] = useState("");
+
+  const noOptions = options.length === 0;
+  const showCreateNewCollection = noOptions || createNewCollection;
+  const description = showCreateNewCollection
+    ? "Lag en samling"
+    : "Velg en samling";
+
+  const handleClose = (): void => {
+    setCreateNewCollection(false);
+    setTextInput("");
+    onClose();
+  };
+
+  const handleCreateNewCollection = (): void => {
+    if (textInput === "") return;
+    if (selectedWordId === null) return;
+
+    addCollection({ title: textInput });
+    setSelected({ label: textInput });
+    addItemToCollection({
+      title: textInput,
+      wordId: selectedWordId,
+    });
+    setTextInput("");
+    onClose();
```

Comment:
I think it should be handleClose(). If not the createNewCollection state persists and next time we click the bookmark icon we get the createCollection dialog again

Reply:
Yes! Thanks for noticing ✨